sdl/sdl.go

```diff
@@ -64,3 +71,20 @@ func Read(buf []byte) (SDL, error) {
 
 	return obj, nil
 }
+
+// Version creates the deterministic Deployment Version hash from the SDL.
+// Sha256 returns 32 byte  sum of the SDL.
+func Version(s SDL) ([]byte, error) {
+	m, err := json.Marshal(s)
```

Comment:
Sorry, I didn't communicate this well - it should be a hash of the manifest -- returned by `s.Manifest()` -- because that is the struct that we send to the providers.

The struct that implements the `SDL` interface is too implementation-dependent - it's only really meant for parsing our particular file, but it should be possible to construct a manifest in many different ways, and the hash of that manifest will be the version regardless of how it was built.

Additionally, since the server has the manifest and not the sdl-parsing object, it can't verify the object we send it.